index.js

```diff
@@ -20,9 +20,13 @@ function scope(closure){
   }
 }
 
-function ps(input){
-  return new scope(function(definition){
-    return patch(definition, input)
+function ps(target, input){
+  return arguments.length === 2
+  ? new scope(function(definition){
+    return patch(target, definition, input)
+  })
+  : new scope(function(definition){
+    return patch(definition, target) // `target` really is `input` in that case
```

Comment:
It took me embarrassingly long to figure out how this worked with such a small patch without breaking anything – I was convinced new functions would be necessary but AFAICT, this simple re-jigging of patch signature accounts for all immutable concerns, right? Incredibly neat if so!

Reply:
I think so.

I'd immediately follow with `psa = input => ps([], input)` and `pso = input => ps({}, input)` in client code.